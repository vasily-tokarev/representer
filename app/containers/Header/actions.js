/*
 *
 * Post actions
 *
 */

import {
  HELP_TOGGLE,
} from './constants';

export function toggleHelp() {
  return {
    type: HELP_TOGGLE,
  };
}

