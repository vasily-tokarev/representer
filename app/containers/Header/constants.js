/*
 *
 * Post constants
 *
 */

export const HELP_TOGGLE = 'HELP_TOGGLE';

