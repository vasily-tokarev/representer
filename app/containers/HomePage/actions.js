/*
 *
 * HomePage actions
 *
 */

import {
  LOAD_POSTS,
  POSTS_LOAD_SUCCESS,
} from './constants';

export function loadPosts() {
  return {
    type: LOAD_POSTS,
  };
}

export function postsLoaded(posts) {
  return {
    type: POSTS_LOAD_SUCCESS,
    posts,
  };
}
