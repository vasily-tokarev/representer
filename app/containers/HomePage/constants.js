/*
 *
 * HomePage constants
 *
 */

export const LOAD_POSTS = 'LOAD_POSTS';
export const POSTS_LOAD_SUCCESS = 'POSTS_LOAD_SUCCESS';
