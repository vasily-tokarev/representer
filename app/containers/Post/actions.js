/*
 *
 * Post actions
 *
 */

import {
  LOAD_POST_SUCCESS,
  LOAD_POST,
} from './constants';

export function loadPost(name) {
  return {
    type: LOAD_POST,
    name,
  };
}

export function postLoaded(post) {
  return {
    type: LOAD_POST_SUCCESS,
    post,
  };
}
