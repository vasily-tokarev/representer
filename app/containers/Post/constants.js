/*
 *
 * Post constants
 *
 */

// export const DEFAULT_ACTION = 'app/Post/DEFAULT_ACTION';
export const LOAD_POST_SUCCESS = 'LOAD_POST_SUCCESS';
export const LOAD_POST = 'LOAD_POST';
