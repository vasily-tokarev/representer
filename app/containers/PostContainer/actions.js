/*
 *
 * Post actions
 *
 */

import {
  LOAD_POST_SUCCESS,
  LOAD_POST,
  UNMOUNT_POST,
  WS_PAYLOAD,
} from './constants';

export function loadPost(name) {
  return {
    type: LOAD_POST,
    name,
  };
}

export function unmountPost() {
  return {
    type: UNMOUNT_POST,
  };
}

export function postLoaded(post) {
  return {
    type: LOAD_POST_SUCCESS,
    post,
  };
}

export function WSPayload(data) {
  return {
    type: WS_PAYLOAD,
    data,
  };
}
