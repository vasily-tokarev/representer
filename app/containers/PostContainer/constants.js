/*
 *
 * Post constants
 *
 */

export const LOAD_POST_SUCCESS = 'LOAD_POST_SUCCESS';
export const LOAD_POST = 'LOAD_POST';
export const UNMOUNT_POST = 'UNMOUNT_POST';

