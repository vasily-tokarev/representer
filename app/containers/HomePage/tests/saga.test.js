/**
 * Test  sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { takeLatest, put } from 'redux-saga/effects';
import postsData, { getPosts } from '../saga';
import { LOAD_POSTS } from '../constants';
import { postsLoaded } from '../actions';

describe('getPosts Saga', () => {
  let getPostsGenerator;
  beforeEach(() => {
    getPostsGenerator = getPosts();

    const callDescriptor = getPostsGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the postsLoaded action', () => {
    const response = [{
      name: 'First post',
    }, {
      name: 'Second post',
    }];
    const putDescriptor = getPostsGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(postsLoaded(response)));
  });
});


describe('postsData Saga', () => {
  const postsDataSaga = postsData();

  it('should start task to watch for LOAD_POSTS action', () => {
    const takeLatestDescriptor = postsDataSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(LOAD_POSTS, getPosts));
  });
});
