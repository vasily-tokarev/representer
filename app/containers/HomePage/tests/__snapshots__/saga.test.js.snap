// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPosts Saga should dispatch the postsLoaded action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "./index.json",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
